src/test/java/redis/clients/jedis/EndpointConfig.java

```diff
@@ -0,0 +1,98 @@
+package redis.clients.jedis;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import redis.clients.jedis.util.JedisURIHelper;
+
+import java.io.FileReader;
+import java.net.URI;
+import java.util.*;
+
+public class EndpointConfig {
+
+    private boolean tls;
+    private String username;
+    private String password;
+    private int bdbId;
+    private Object rawEndpoints;
+    private List<URI> endpoints;
+
+
+    public EndpointConfig(HostAndPort hnp, String username, String password, boolean tls) {
+        this.tls = tls;
+        this.username = username;
+        this.password = password;
+        this.bdbId = 0;
+        this.rawEndpoints = null;
+        this.endpoints = Collections.singletonList(
+                URI.create(getURISchema() + hnp.getHost() + ":" + hnp.getPort())
+        );
+    }
+
+    public HostAndPort getHostAndPort() {
+        return JedisURIHelper.getHostAndPort(endpoints.get(0));
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getHost() {
+        return getHostAndPort().getHost();
+    }
+
+    public int getPort() {
+        return getHostAndPort().getPort();
+    }
+
+    public URI getURI() {
+        return endpoints.get(0);
+    }
+
+    public URI getCustomizedURI(boolean withCredentials, String path) {
```

Comment:
Would be great to expose a customized URI builder, where the caller can add things like `defaultCredentials()`, `customCredentials(u, p)`, `customPath(p)` and so on. Otherwise we might end up with more and more parameters on these methods, which is harder to read/maintain in the tests.

Reply:
Good point! Done.